### PR TITLE
POSIX backtrace fixes

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: Infrastructure "POSIX backtraces now work for installed binaries and position-independent executables"

#### Purpose of change
There are a couple of issues with the existing POSIX backtrace symbolizing:
* It didn't work when cataclysm was run and found via `$PATH`, rather than e.g. `./cataclysm`.
* It didn't work when ASLR was in effect.

#### Describe the solution
* Perform our own `PATH` lookup to find the binary when we were not given the location.
* Use `nm` to calculate the ASLR load offset and thereby pass the correct addresses to `addr2line`.

#### Describe alternatives you've considered
This approach using `nm` is rather hacky, and I'm not sure how portable it is.  However, the alternatives I considered are definitely not portable.  At least some standard libraries (including glibc) offer custom APIs to determine the ASLR load offset, but there seemed to be little similarity between them.

The neatest solution would be to abandon this effort to roll our own solution and use libbacktrace instead, but that adds another dependency.

Yet another option is to disable position-independence when linking.  This is awkward because the options depend on compiler version.  Also, it only fixes the issue for `cataclysm` itself, not any libraries.  And I believe Android is moving to requiring executables be position-independent.

#### Additional context
I added `string_starts_with` and `string_ends_with` to `cata_utility.hpp`.

### CODE REVIEW

File: src/debug.cpp [hunk 1]
```diff
@@ -446,6 +447,92 @@ bool debug_is_safe_string( const char *start, const char *finish )
...
+            continue;
+        }
+        std::string candidate = path_elem + "/" + binary;
+        if( 0 == access( candidate.c_str(), X_OK ) ) {
+            return candidate;
+        }
+    }
+
+    return binary;
+};
```
Comment (BevapDin) [inline src/debug.cpp]: That semicolon is not needed.

Comment (jbytheway) [reply src/debug.cpp]: Thanks, good catch.

**Comment by jbytheway** [comment]
Looks good now.

Note that the Jenkins is never going to pass because I've changed the top-level Makefile.

Here you can see a new screenshot from CI:
cata-backtrace-2
Note that we now have some details of the functions inside libc.so.6, which was not working before.

Mingw backtraces are not working.  I feel like they did work when I added backtraces-in-tests originally, but I checked and they're also not working on master at the moment, so this PR is not what broke them.

**Comment by ZhilkinSerg** [comment]
```
11:24:49 +./mainline-build.zsh:101> make clean
11:24:49 Makefile:775: *** missing separator.  Stop.
11:24:49 Build step 'Conditional step (single)' marked build as failure
11:24:49 SSH: Current build result is [FAILURE], not going to run.
```

File: Makefile [hunk 2]
```diff
@@ -768,7 +772,7 @@ $(TARGET): $(OBJS)
 ifdef RELEASE
   ifndef DEBUG_SYMBOLS
     ifneq ($(BACKTRACE),1)
-	$(STRIP) $(TARGET)
+      $(STRIP) $(TARGET)
```
Comment (ZhilkinSerg) [inline Makefile]: It seems something is not right here.

Comment (jbytheway) [reply Makefile]: Yep, my bad; that has to remain a tab, not spaces.  Sorry, didn't test in release build.

**Pull request merged**